base/array.jl

```diff
@@ -2612,3 +2612,10 @@ end
 
 intersect(itr, itrs...) = _shrink(intersect!, itr, itrs)
 setdiff(  itr, itrs...) = _shrink(setdiff!, itr, itrs)
+
+function intersect(v::AbstractVector, r::AbstractRange)
+    common = Iterators.filter(x -> x ∈ r, v)
+    seen = Set{eltype(v)}(common)
```

Comment:
Is there a reason to do it this way instead of just `filter(in(r), v)`?

Reply:
Ah I see, it's to get unique values.

Reply:
Yes, `seen` contains unique elements. It is then used in the `_shrink_filter!` below, therefore the name `seen`. Should I put a short comment there?